celery/app/base.py

```diff
@@ -471,6 +477,18 @@ def run(*args, **kwargs):
                     try:
                         return task._orig_run(*args, **kwargs)
                     except autoretry_for as exc:
+                        if retry_backoff:
```

Comment:
The more compact form could potentially be (I think the `if` block is not required):
```
countdown = retry_backoff * (2 ** task.request.retries) # will be zero if retry_backoff = 0
countdown += random.randrange(retry_jitter + 1) * random.choice((1, -1)) # if retry_jitter is 0, the result is always zero too
# Adjust according to maximum backoff
countdown = min(retry_backoff_max, countdown)
```

- `random.randrange` excludes the endpoint that is why I added `1`. This was also excluding defining `1` as a possible jitter value.
- If `retry_jitter` is 0, `countdown` remains the same.
- If `retry_jitter` is 2, and delay is set to 8, then, possible random values are `[0, 1, 2]`, and `countdown` will be in the range `[6, 10]`.